### ISSUE TITLE
Blocks move when a project is reloaded

### ISSUE THREAD

**Issue by martinwork** [microsoft/pxt-microbit#5947]
**Describe the bug**

Arising from support ticket https://support.microbit.org/helpdesk/tickets/82439 (private)

After creating a project with blocks neatly aligned, the blocks move in an apparently unpredictable way when the project is reloaded.

The movement seems to vary - does it depend on which block was moved last?

The moving blocks move across more on each reload.

I don't know if functions are essential - but the example in the ticket has functions.

**To Reproduce**
1. Go to https://makecode.microbit.org/_hLc2Ve1zYbEz
2. Position blocks as below
3. Visit Home and reload the project

**Expected behavior**
Blocks don't change position.

**Screenshots**
Before reload
![image](https://github.com/user-attachments/assets/00125804-9062-4b4e-befd-a7f84ebd5653)

After reload
![image](https://github.com/user-attachments/assets/e0277a59-87e4-4111-ac77-0a3d595d9517)

**micro:bit version (please complete the following information):**
not hardware related

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] 128

**Comment by martinwork** [comment]
Related to this, though not a regression, I don't think... Should I create a separate issue?

Create a new project
![image](https://github.com/user-attachments/assets/b50365ed-2015-499a-b11f-b2e4e52c409d)

Reload it - border is smaller
![image](https://github.com/user-attachments/assets/9f83b134-e436-4129-b3a9-58de3c6e89bc)

===========================

Sometimes there's no border...
![image](https://github.com/user-attachments/assets/08fc83ba-7d7d-4577-8372-a99ec3731e1d)

becomes
![image](https://github.com/user-attachments/assets/6658a553-630e-409e-9dd2-0cfc914f4b1e)

==========================

Or a border on one side...
![image](https://github.com/user-attachments/assets/c69cd1d7-a318-485d-95d0-09d243d05e86)

becomes
![image](https://github.com/user-attachments/assets/11cd964d-e951-41f8-98e8-2469be168d35)

**Comment by riknoll** [comment]
i've run into this; seems like the blockly xml loader now likes to bump blocks when parsing the xml file which causes these little nudges to happen.

the behavior in your second comment, however, is how it's always been iirc. we've always adjusted the workspace to align everything to the top-left when loading a project

**Comment by martinwork** [comment]
Thanks @riknoll I suppose users want to lay out blocks in neat columns and rows and not have that undone with overlapping blocks when the project is reloaded.

The second part is not new, but I noticed it again while looking at the first thing and thought it might be related. I had two thoughts:
- the border of a new project and a reloaded project should be the same width
- no border on either or both sides doesn't look right

**Comment by riknoll** [comment]
i agree that the nudging behavior is something we should fix, but i believe that we made the aligning to the top-left change in response to feedback from user testing. people were getting "lost" when they opened large projects iirc

**Comment by martinwork** [comment]
OK, I don't feel strongly about the the border behaviour, which has clearly stood the test of time!

**Comment by martinwork** [comment]
@riknoll But I don't understand why this issue about blocks moving on reload was removed from the hotfix milestone. Was that a mistake because I've confused the issue?

**Comment by martinwork** [comment]
Also reported in support ticket https://support.microbit.org/helpdesk/tickets/83796 (private) that, paraphrasing, when the page is reloaded, blocks move around and often end up overlapping.

**Comment by abchatra** [comment]
This is fixed in /beta. @martinwork can you validate?

**Comment by martinwork** [comment]
Ah! It seems I have not received the latest beta yet.

**Comment by abchatra** [comment]
Please open new issue for this.

**Comment by martinwork** [comment]
@abchatra OK. I'll test it again when this week's changes are released to beta.

**Comment by martinwork** [comment]
This is fixed in 
makecode.microbit.org version:  [7.1.33](https://github.com/Microsoft/pxt-microbit/releases/tag/v7.1.33)
Microsoft MakeCode version:  [11.4.21](https://github.com/Microsoft/pxt/releases/tag/v11.4.21)
